### PR TITLE
Generalize flakefinder for future gerrit usage

### PR DESCRIPTION
**Pull request by rmohr**
Move github structs and clients behind interfaces which allow working on gerrit and gerrit Change Requests in the future as well.

* `PullRequest` is now hidden behind an `api.Change` interface
* GitHub client is hidden behind a `api.Query` interface

Checking if a prow job was executed on the latest change was moved behind the `api.Query` interface.

### CODE REVIEW

File: robots/pkg/flakefinder/flakefinder.go
Comment (dhiller) [inline robots/pkg/flakefinder/flakefinder.go:197]: nit
```suggestion
	for _, change := range changes {
```

File: robots/pkg/flakefinder/downloader_test.go
Comment (dhiller) [inline robots/pkg/flakefinder/downloader_test.go:7]: nit: I find the `github2` ugly and a bit confusing, since the reader has to keep in mind that they are looking at usage of the GitHub implementation for the "repo query api" so to say. I understand that you want to keep it consistent though.

So, since I can't come up with a better solution for this right now myself I guess this improvement is for a later PR.

**Review by dhiller — APPROVED** [review]
Just two nits inline, in general these changes look fine to me. I'll leave it up to you if you want to address them. Just unhold in the latter case.

/approve
/hold

File: robots/pkg/flakefinder/downloader_test.go
Comment (rmohr) [reply robots/pkg/flakefinder/downloader_test.go]: Done. Renamed to `ghapi`.

File: robots/pkg/flakefinder/flakefinder.go
Comment (rmohr) [reply robots/pkg/flakefinder/flakefinder.go]: Done.

**Comment by rmohr** [comment]
/unhold

PTAL

**Comment by dhiller** [comment]
> /unhold
> 
> PTAL

@rmohr was out for a couple of days, will take another look soon!

**Review by dhiller — APPROVED** [review]
@rmohr looks great, thanks!

**Pull request merged**
